### DISCUSSION TITLE
Project Schema w/ validation

### DISCUSSION THREAD

**Discussion by kpervin** [Infisical/infisical#5708] [Feature Request]
### Feature description
As a user, I would like to set a schema across a project to ensure that every environment includes the correct variables and are of the correct "type". 

Additionally, having the ability to do conditional validation against the variable values themselves would be amazing.

### Why would it be useful?
This would be ideal as it would prohibit saving an environment unless all variables are set, thus failing fast before any sort of deploy occurs.

### Additional context
Using something along the lines of [Joi](https://joi.dev/) or [Zod ](https://zod.dev/) for validation rules, or even a custom GUI option, would be great to harness greater specificity in a project.

**Comment by kpervin** [comment]
Alternatively, something along [Doppler's Templates](https://docs.doppler.com/docs/project-templates) would be a great feature as well

**Comment by dangtony98** [comment]
Thanks for this feature suggestion @kpervin!

We’ve been thinking about validation for environment variables as well; there’s definitely opportunity here, though we haven’t figured out the optimal approach.

We can jam on the specific implementation for this more 😄
